### PR TITLE
lndclient/basic_client: pass full macaroon path, not just directory.

### PR DESCRIPTION
**Pull request by valentinewallace**
This is useful for lndmon's usage of loop's lndclient package so we can restrict lndmon to `readonly.macaroon` rather than `admin.macaroon`.

### CODE REVIEW

File: lndclient/basic_client.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ import (
 // "basic" as it uses a global macaroon (by default the admin macaroon) for the
 // entire connection, and falls back to expected defaults if the arguments
 // aren't provided.
-func NewBasicClient(lndHost, tlsPath, macDir, network string) (lnrpc.LightningClient, error) {
+func NewBasicClient(lndHost, tlsPath, macPath, network string) (lnrpc.LightningClient, error) {
```
Comment (Roasbeef) [inline lndclient/basic_client.go]: I think we can avoid changing the set of arguments too much by using a functional argument here. This path will also mean that callers won't need to update their set of arguments they want the read only macaroon (admin being the default for now, though we should revisit, least authority principle n all).

Comment (valentinewallace) [reply lndclient/basic_client.go]: switched it to take in a `readonly` boolean param.

Comment (valentinewallace) [reply lndclient/basic_client.go]: ok updated to use functional optional args.

**Comment by Roasbeef** [comment]
Ping!

**Comment by valentinewallace** [comment]
> Ping!

Updated!

**Pull request merged**